# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: third_party/rust/gpu-descriptor/src/lib.rs
# Commit: ef5dc3e04e5f
# Full Hash: ef5dc3e04e5f271eea0636ab3a495e95cc912f1d
# Author: Dzmitry Malyshau <dmalyshau@mozilla.com>
# Date: 2021-09-04 09:40:24
# Regressor Bug: 1726626
# File Overlap Count: 3
# Description:
#   Bug 1726626 - Move gfx/wgpu into a 3rd party dependency r=jgilbert,bholley
#   
#   This update makes wgpu a vendored dependency instead of having it in gfx/wgpu.
#   
#   ## Notes
# ==============================================================================

diff --git a/third_party/rust/gpu-descriptor/src/lib.rs b/third_party/rust/gpu-descriptor/src/lib.rs
--- a/third_party/rust/gpu-descriptor/src/lib.rs
+++ b/third_party/rust/gpu-descriptor/src/lib.rs
@@ -1,10 +1,26 @@
 //!
-//! gpu-descriptor crate.
+//! Library for Vulkan-like APIs to allocated descriptor sets
+//! from descriptor pools fast, with least overhead and zero fragmentation.
+//!
+//! Straightforward usage:
+//! ```ignore
+//! use gpu_descriptor::DescriptorAllocator;
+//!
+//! let mut allocator = DescriptorAllocator::new(max_update_after_bind_descriptors_in_all_pools); // Limit as dictated by API for selected hardware
+//!
+//! let result = allocator.allocate(
+//!     device, // Implementation of `gpu_descriptor::DescriptorDevice`. Comes from plugins.
+//!     layout, // Descriptor set layout recognized by device's type.
+//!     flags,  // Flags specified when layout was created.
+//!     layout_descriptor_count, // Descriptors count in the layout.
+//!     count, // count of sets to allocated.
+//! );
+//! ```
 //!
 
 #![cfg_attr(not(feature = "std"), no_std)]
 #![warn(
     missing_docs,
     trivial_casts,
     trivial_numeric_casts,
     unused_extern_crates,